src/main/java/uk/nhs/hee/tis/trainee/sync/config/MongoConfiguration.java

```diff
@@ -51,5 +52,13 @@ public void initIndexes() {
     IndexOperations placementIndexOps = template.indexOps(Placement.class);
     placementIndexOps.ensureIndex(new Index().on("data.postId", Direction.ASC));
     placementIndexOps.ensureIndex(new Index().on("data.siteId", Direction.ASC));
+
+    IndexOperations programmeMembershipIndexOps = template.indexOps(ProgrammeMembership.class);
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.programmeId", Direction.ASC));
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.curriculumId", Direction.ASC));
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.personId", Direction.ASC));
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.programmeMembershipType", Direction.ASC));
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.programmeStartDate", Direction.ASC));
+    programmeMembershipIndexOps.ensureIndex(new Index().on("data.programmeEndDate", Direction.ASC));
```

Comment:
I don't know Mongo indexing but my assumption (from SQL DBs) is that a single index would be used for `findByPersonIdAndProgrammeIdAndProgrammeMembershipTypeAndProgrammeStartDateAndProgrammeEndDate()` and I can't see where else they'd be used.

Reply:
That's true. I've pushed a change to implement this. I think I had left them as a set of individual indexes because some of them (programmeId, curriculumId, personId) are used individually in ProgrammeMembershipRepository.